# Computional Genomics WorkShop
Instructions and assignments for the computational genomics workshop given at RUNI 2025

## General information and requirments

 **Don’t forget to bring your laptop to class**

We start our exploration by getting deeply familiar with the most predominant method of sequencing to date, Next Generation Sequencing (NGS). We will learn how high-throughput sequencing is performed and get our computational tools configured in order to search for novel cancerous mutations in the genome of a cancer patient.

For our mission to be successful we trust you already have your Unix-like OS ready, preferably a Linux distro. MacOS users should be fine to install all software and Windows users are encouraged to enable [WSL](https://learn.microsoft.com/en-us/windows/wsl/install) and install one of the Linux distributions offered on the Microsoft Store.

To facilitate easy installation of bioinformatic tools please consider installing the Anaconda package manager (the miniconda variant is fine) and configuring the bioconda channel. A detailed (short and simple) guide on how to achieve it can be found [here](https://bioconda.github.io/). Class and home work will assume a functional conda environment, but tech savvy students should be fine with whatever method they choose to get binaries for their system. Please make sure you have ~20Gb of available storage on your machine, mainly for the sequencing data and auxiliary files.

In the following weeks class time will be split into two sections, we will start with a lecture followed by hands-on time for an assignment you’d have to submit up to a week after, so don’t forget to bring your laptop to class.

amit.levon@post.runi.ac.il


